geode-docs/basic_config/data_entries_custom_classes/managing_data_entries.html.md.erb

```diff
@@ -60,36 +60,90 @@ You can also use the `gfsh put` command to add entries to a region, and the `get
+they were placed in the `Map`.
+For partitioned regions,
+multiple events are sent as a single message to the primary buckets
+and then distributed to the secondary buckets.
+
+The design of a client application within a client-server design pattern
+faces the possibility that a partial operation can occur.
+Some, all, or none of the specified entries may be completed with `putAll`.
+If either `ServerOperationException` or `ServerConnectivityException` are
+thrown,
+it can indicate an incomplete operation.
 
 ``` pre
-void putAll(String command) throws CacheException 
-{ 
-// Get Entry keys and values into Strings key1, ... keyN and value1, ... valueN 
-  Map map = new LinkedHashMap(); 
-  map.put(key1, value1)); 
-  ...
-  map.put(keyN, valueN));
-  this.currRegion.putAll(map); 
+int retry = 0;
+RuntimeException rte = null;
+while (retry < 3) {
+  try {
+    region.putAll(map);
+  } catch (ServerOperationException e) {
+    if (e.getCause() instanceof TimeoutException
+        || e.getCause() instanceof LowMemoryException) {
+      // Retry due to transient error.
+      retry++;
+    } else {
+      rte = e;
+      break;
+    }
+  } catch (ServerConnectivityException e) {
+    // Retry due to transient error.
+    retry++;
+  }
+}
+
+if (retry == 3) {
+  System.out.println("3 putAll operations tried, and all failed.");
+} else if (rte != null) {
+  System.out.println("putAll failed due to " + rte);
+  throw rte;
 }
 ```
 
+A thrown exception of `ServerConnectivityException` with a cause of
+`TimeoutException` or `LowMemoryException` can be a transient error.
+A limited quantity of retries of the `putAll` may result in a completed
+operation.
+A repeated timeout may imply that the `read-timeout` value is not
+long enough to complete the bulk operation;
+use the `org.apache.geode.cache.client.PoolFactory.setReadTimeout`
+method to set the `read-timeout` value.
+
+Client applications that cannot tolerate partial completion of a `putAll`
+operation may embed the operation into a transaction.
+See [Transactions](../../developing/transactions/chapter_overview.html)
+for details.
+
+**Note:**
+The processing of a map with many entries and/or extra-large data values
+may affect system performance and cause cache update timeouts,
+especially if the region uses overflow or persistence to disk.
+
+## <a id="removeAll_method" class="no-quick-link"></a>The removeAll Operation
+
+The batch operation `Region.removeAll`
+
 The updates to the cache are done individually in the order in which they were placed in the `Map`. For partitioned regions, multiple events are sent as a single message to the primary buckets and then distributed to the secondary buckets.
 
 **Note:**
-The processing of maps with very many entries and/or very large data may affect system performance and cause cache update timeouts, especially if the region uses overflow or persistence to disk.
+The processing of a map with many entries and/or extra-large data values
```

Comment:
Line 144-146 looks like duplicate of line 133-135.

Reply:
Purposeful repeat. One of these is for putAll, and the other is for removeAll. Thank for your attention to detail!